public/app/features/integrated-alerting/components/AlertRules/AddAlertRuleModal/AddAlertRuleModal.tsx

```diff
@@ -0,0 +1,159 @@
+  LoaderButton,
+  TextInputField,
+  NumberInputField,
+  TextareaInputField,
+  logger,
+  validators,
+} from '@percona/platform-core';
+import { SelectableValue } from '@grafana/data';
+import { AppEvents } from '@grafana/data';
+import { Messages } from './AddAlertRuleModal.messages';
+import { AddAlertRuleModalProps, AddAlertRuleFormValues } from './AddAlertRuleModal.types';
+import { getStyles } from './AddAlertRuleModal.styles';
+import { SEVERITY_OPTIONS } from './AddAlertRulesModal.constants';
+import { formatTemplateOptions, formatChannelsOptions, formatCreateAPIPayload } from './AddAlertRuleModal.utils';
+import { AlertRulesProvider } from '../AlertRules.provider';
+import { AlertRulesService } from '../AlertRules.service';
+import { AlertRuleTemplateService } from '../../AlertRuleTemplate/AlertRuleTemplate.service';
+import { NotificationChannelService } from '../../NotificationChannel/NotificationChannel.service';
+import { appEvents } from 'app/core/core';
+
+const { required } = validators;
+
+export const AddAlertRuleModal: FC<AddAlertRuleModalProps> = ({ isVisible, setVisible }) => {
+  const styles = useStyles(getStyles);
+  const [templateOptions, setTemplateOptions] = useState<Array<SelectableValue<string>>>();
+  const [channelsOptions, setChannelsOptions] = useState<Array<SelectableValue<string>>>();
+  const { getAlertRules } = useContext(AlertRulesProvider);
+
+  const getData = async () => {
+    try {
+      const channelsListResponse = await NotificationChannelService.list();
+      setChannelsOptions(formatChannelsOptions(channelsListResponse));
+      const templatesListResponse = await AlertRuleTemplateService.list();
+      setTemplateOptions(formatTemplateOptions(templatesListResponse.templates));
+    } catch (e) {
+      logger.error(e);
+    }
+  };
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+  const onSubmit = async (values: AddAlertRuleFormValues) => {
+    try {
+      await AlertRulesService.create(formatCreateAPIPayload(values));
+      setVisible(false);
+      appEvents.emit(AppEvents.alertSuccess, [Messages.addSuccess]);
+      getAlertRules();
+    } catch (e) {
+      logger.error(e);
+    }
+  };
+
+  return (
+    <Modal
+      title={Messages.title}
+      isVisible={isVisible}
+      onClose={() => setVisible(false)}
+      data-qa="add-alert-rule-modal"
+    >
+      <Form
+        onSubmit={onSubmit}
+        render={({ handleSubmit, valid, pristine, submitting }) => (
+          <form className={styles.form} onSubmit={handleSubmit} data-qa="add-alert-rule-modal-form">
+            {/* TODO: polish this up */}
+            <Field name="template" validate={required}>
+              {({ input }) => (
+                <>
+                  <label className={styles.label} data-qa="type-field-label">
+                    {Messages.templateField}
+                  </label>
+                  <Select className={styles.select} options={templateOptions} {...input} data-qa="template-select" />
```

Comment:
Other inputs come with `<name>-<type>-input`, so maybe it makes sense to do the same here, i.e. `template-select-input`?

Reply:
Done